libverify/verify_libfuzzer.cpp

```diff
@@ -0,0 +1,834 @@
+/**
+ * Defines an interactive implementation of assume for libfuzzer. When
+ * assumption fails, the reason is logged, and complete the requirements
+ * @date 2019
+ */
+
+#include "verify.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <stdint.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <assert.h>
+
+/**
+ * Defines an interface to libverify, with reconfigurable data types. These
+ * types are controlled at build time using preprocessor definitions.
+ */
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*global variable Env record the environment*/
+static jmp_buf Env;
+
+/*global variable exception_type 
+ * record the return value of setjmp function,
+ * which can label different exception types*/
+static int exception_type;
```

Comment:
Minor nitpick. I think this should be an enum. Something like...

```
enum ExceptionType { NONE, OUT_OF_DATA, REQUIRE_FAILED };
static ExceptionType exception_type = NONE;
```

This will be self-documenting, easier to refactor, and less error prone.

Reply:
ok~